middlewares/checkIfUserAllowed.js

```diff
@@ -0,0 +1,73 @@
+const { ROLE_NAME, RESOURCE } = require("../constants/index.js");
+const Comment = require("../models/commentModel");
+const Post = require("../models/postModel.js");
+const User = require("../models/userModel.js");
+
+const checkIfUserAllowed = (resource) => async (req, res, next) => {
+  try {
+    const id = req.params.id;
+    if (resource === RESOURCE.User) {
+      if (
+        req.method === "DELETE" ||
+        ("PUT" &&
```

Comment:
this will not work properly, you just comparing "PUT" string, but you should check if req.method === "PUT"